doc/dev/testing/running-the-tests.md

```diff
@@ -0,0 +1,129 @@
+# Running the SDK Tests
+
+The operator-sdk is automatically tested with a variety of tests anytime
+a pull request is made. The E2E tests ensure that the operator-sdk acts as intended by
+simulating how a typical user might use the SDK. The automated tests test each PR and run in
+Travis CI, and Travis CI has a couple of features to simplify the E2E tests that we run. For
+a more in depth description of the tests that Travis runs, please read the [Travis Build][travis] doc.
+This doc will talk about how to run the tests locally as well.
+
+## Running the Tests Locally
+
+To run the tests locally, the tests either need access to a remote kubernetes cluster or a
+local kubernetes instance running on the machine.
+
+### Remote Kubernetes Instance
+
+To run the tests on a remote cluster, the tests need access to a remote kubernetes cluster
+running kubernetes 1.11 as well as a docker image repo to push the operator image to,
+such as [`quay.io`][quay]. Your kubeconfig must be located at `$HOME/.kube/config` and certain
+tests will not run on remote clusters. See [Running the Tests](#running-the-tests) for more details.
+
+### Local Openshift Cluster
+
+One way to run the tests is with an openshift 3.11 cluster and `oc cluster up` on a Linux system.
+
+For the first run configuration, either you can either run the `hack/ci/setup-openshift.sh` script, or download
+and install the [oc binary][oc-binary] and run these commands:
+
+```sh
+$ sudo service docker stop
+$ sudo sed -i 's/DOCKER_OPTS=\"/DOCKER_OPTS=\"--insecure-registry 172.30.0.0\/16 /' /etc/default/docker
+$ sudo service docker start
+```
+
+Depending on the system you are on, you may also need to configure your firewall before running the cluster.
+Refer to the [official docs][oc-docs] for more information.
+
+After the initial configuration, you can use this command to start the cluster and login as admin:
+
+```sh
+$ oc cluster up --base-dir=$HOME/oscluster
+$ oc login -u system:admin
+```
+
+You can use this command to stop the cluster:
+
+```sh
+$ oc cluster down
+```
+
+To fully delete the cluster, you must run these commands:
+
+```sh
+$ oc cluster down
+$ sudo umount $(grep $HOME/oscluster /proc/mounts | cut -f2 -d" " | sort -r)
+$ sudo rm -rf $HOME/oscluster
+```
+
+**NOTE**: Starting with openshift 4.0, the `oc cluster up` command will not be availble. This document
+and the tests will be updated in the future to support openshift 4.0.
+
+
+### Local Minikube
+
+Another option for testing is using minikube. This is not advised as it uses vanilla kubernetes, which has less
+strict security and may allow some tests to pass when they would not under openshift. Minikube is faster than
+openshift and uses less RAM though. To start the minikube cluster, download and install the proper [binary][minikube-binary]
+for your system and run these commands.
+
+```sh
+$ minikube start
```

Comment:
You may want to add the version flag here, you must use greater than 1.11 and the default is still 1.10 I think.